packages/gatsby-plugin-google-tagmanager/src/gatsby-ssr.js

```diff
@@ -27,7 +41,8 @@ exports.onRenderBody = (
             new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
             j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
             'https://www.googletagmanager.com/gtm.js?id='+i+dl+'${environmentParamStr}';f.parentNode.insertBefore(j,f);
-            })(window,document,'script','dataLayer', '${pluginOptions.id}');`,
+            })(window,document,'script','${pluginOptions.dataLayerName ||
```

Comment:
In context of the comment above, this should be enough according to https://developers.google.com/tag-manager/devguide

Reply:
That's correct. I pushed my changes.